plugin/pkg/scheduler/factory/factory.go

```diff
@@ -448,6 +449,19 @@ func (b *binder) Bind(binding *api.Binding) error {
 	// return b.Pods(binding.Namespace).Bind(binding)
 }
 
+type podConditionUpdater struct {
+	*client.Client
+}
+
+func (p *podConditionUpdater) Update(pod *api.Pod, condition *api.PodCondition) error {
+	glog.V(2).Infof("Updating pod condition for %s/%s to (%s==%s)", pod.Namespace, pod.Name, condition.Type, condition.Status)
+	if api.UpdatePodCondition(&pod.Status, condition) {
+		_, err := p.Pods(pod.Namespace).UpdateStatus(pod)
```

Comment:
AFAIK, the client is doing some retries:
https://github.com/kubernetes/kubernetes/blob/8472cfa214369f69f2dc680962be63ba34199bd2/pkg/client/restclient/request.go#L785-L785

I'm inclined to make it async. But not super worried about it because of error path. I will let @davidopp make the decision.